### PR TITLE
Add fake container registry component for Wasm pulling integration test.

### PR DESCRIPTION
**Pull request by bianpengyuan**
**Please provide a description of this PR:**

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/test/framework/components/containerregistry/container_registry_server.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    matchLabels:
+      app: container-registry
+  template:
+    metadata:
+      labels:
+        app: container-registry
+    spec:
+      containers:
+      - image: gcr.io/istio-testing/fake-registry:1.0
+        imagePullPolicy: Always
```
Comment (howardjohn) [inline pkg/test/framework/components/containerregistry/container_registry_server.yaml:43]: Probably no need to use always? we should use immutable tags

File: pkg/test/framework/components/containerregistry/container_registry_server.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        name: container-registry
+        ports:
```
Comment (howardjohn) [inline pkg/test/framework/components/containerregistry/container_registry_server.yaml:45]: can we add readiness probe

Comment (bianpengyuan) [reply pkg/test/framework/components/containerregistry/container_registry_server.yaml:44]: Yeah sorry somehow missed it.

**Comment by bianpengyuan** [comment]
@howardjohn mind approve this?

**Review by howardjohn — APPROVED** [review]

**Pull request merged**